workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowViewStub.kt

```diff
@@ -210,36 +213,30 @@ public class WorkflowViewStub @JvmOverloads constructor(
     rendering: Screen,
     viewEnvironment: ViewEnvironment
   ) {
-    holder?.takeIf { it.canShow(rendering) }
-      ?.let {
-        it.show(rendering, viewEnvironment)
-        return
-      }
+    val oldViewOrNull = showing.currentVisualOrNull
 
-    val parent = actual.parent as? ViewGroup
-      ?: throw IllegalStateException("WorkflowViewStub must have a non-null ViewGroup parent")
-
-    holder?.view?.let {
-      // The old view is about to be detached by replaceOldViewInParent. When that happens,
-      // it's not just a regular detach, it's a navigation event that effectively says that view
-      // will never come back. Thus, we want its Lifecycle to move to permanently destroyed, even
-      // though the parent lifecycle is still probably alive.
-      WorkflowLifecycleOwner.get(it)?.destroyOnDetach()
-    }
+    showing.replaceWith(rendering, context, viewEnvironment) { newView, doFirstUpdate ->
 
-    holder = rendering.toViewFactory(viewEnvironment)
-      .startShowing(rendering, viewEnvironment, parent.context, parent) { view, doStart ->
-        WorkflowLifecycleOwner.installOn(view)
-        doStart()
-      }.also {
-        val newView = it.view
+      val parent = actual.parent as? ViewGroup
+        ?: throw IllegalStateException("WorkflowViewStub must have a non-null ViewGroup parent")
 
-        if (inflatedId != NO_ID) newView.id = inflatedId
-        if (updatesVisibility) newView.visibility = visibility
-        background?.let { newView.background = it }
-        propagateSavedStateRegistryOwner(newView)
-        replaceOldViewInParent(parent, newView)
+      oldViewOrNull?.let {
+        // The old view is about to be detached by replaceOldViewInParent. When that happens,
+        // it's not just a regular detach, it's a navigation event that effectively says that view
+        // will never come back. Thus, we want its Lifecycle to move to permanently destroyed, even
+        // though the parent lifecycle is still probably alive.
+        WorkflowLifecycleOwner.get(it)?.destroyOnDetach()
       }
+      WorkflowLifecycleOwner.installOn(newView)
+      doFirstUpdate()
+
+      if (inflatedId != NO_ID) newView.id = inflatedId
+      if (updatesVisibility) newView.visibility = visibility
+      background?.let { newView.background = it }
+      propagateSavedStateRegistryOwner(newView)
+
+      replaceOldViewInParent(parent, newView)
+    }
```

Comment:
I'd eventually aim for making replaceWith receive a `doUpdate = true / false` and use false for this case.
Or have onNewVisualBeforeUpdate / onNewVisualAfterUpdate. 🤷 
Having MultiRendering do the update for us makes sense, but it shouldn't complicate us.

Reply:
Not really following. I _love_ this, providing a lambda that gets called only if there's an update to do. Seems much simpler than what I was doing before.

Reply:
Maybe give the lambda three arguments, build `oldViewOrNull` into it.